### PR TITLE
feat: claim all dialog

### LINKED ISSUE #3033: [Demand] The Claim Your Tokens Dialog
## Catalogue

- [x] UI/UX
- [ ] Mask Core Feature
- [x] Mask Plugin Feature (ITO)
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background

Be able to claim one's owned tokens in multiple ITO rounds.

+ TOKENs are not claimable if those claimed TOKENs were locked in the contract.
+ TOKENs are claimable if the locking period is expired.

### Token Card

It should present a TOKEN CARD component for each claimable or unclaimable TOKENs. A TOKEN with the same address but has a different locking period will render multiple TOKEN CARD components. 

E.g., If an account has got `1` unlocked TOKEN A, with `2` locked TOKEN A in locked period `t1` and `3` locked TOKEN A in locked period `t2`. It should render three TOKEN CARD components.

+ TOKEN A with unlocked amount `1`.
+ TOKEN A with locked amount `2` in expiration time `t1`.
+ TOKEN A with locked amount `3` in expiration time `t2`.

### Claim Button

+ It should be disabled if nothing to claim.
+ It should be enabled if it has got any TOKENs to be claimed.
+ Click this button is meant to claim all of the claimable TOKENs at that time.

## Notes

+ [Figma](https://www.figma.com/file/gVkQ67y285b4FXVV1KPThN/Twitter?node-id=11150%3A268323)

### PR DESCRIPTION
**Pull request by zhouhanseng**
closes #3033

### CODE REVIEW

File: packages/maskbook/src/plugins/ITO/assets/formatDate.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export function datetimeISOString(value: Date): string {
         const pad = (value: number) => value.toString(10).padStart(2, '0')
         const date = [value.getFullYear(), value.getMonth() + 1, value.getDate()]
         const time = [value.getHours(), value.getMinutes(), value.getSeconds()]
-        return `${date.map(pad).join('-')}T${time.map(pad).join(':')}`
+        return `${date.map(pad).join('-')} ${time.map(pad).join(':')}`
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/assets/formatDate.ts:18]: use `date-fns/format` replace it.

Comment (zhouhanseng) [reply packages/maskbook/src/plugins/ITO/assets/formatDate.ts]: done

File: packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx [hunk 2]
```diff
@@ -119,56 +135,47 @@ export function ClaimAllDialog(props: ClaimAllDialogProps) {
...
+                            {swappedTokens.map((swappedToken, i) => (
+                                <ListItem key={i} className={classes.tokenCard}>
+                                    <div className={classes.cardHeader}>
+                                        <Typography>
+                                            {swappedToken.token.symbol}{' '}
+                                            {swappedToken.isClaimable ? 'Unclaimed' : 'Locked'}:
+                                        </Typography>
+                                        {swappedToken.isClaimable ? null : (
+                                            <Typography>
+                                                Unlock Time:{' '}
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx:159]:
```suggestion
                                                <span>Unlock Time: </span>
```

Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx:159]: why not i18n?

File: packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx [hunk 3]
```diff
@@ -152,11 +152,14 @@ export function ClaimAllDialog(props: ClaimAllDialogProps) {
...
+                                            {swappedToken.isClaimable
+                                                ? t('plugin_ito_claim_all_status_unclaimed')
+                                                : t('plugin_ito_claim_all_status_locked')}
+                                            :
                                         </Typography>
                                         {swappedToken.isClaimable ? null : (
                                             <Typography>
-                                                Unlock Time:{' '}
+                                                {t('plugin_ito_claim_all_unlock_time')}:{' '}
                                                 {formatDateTime(swappedToken.unlockTime, 'yyyy-MM-dd hh:mm:ss')}
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx]:
```suggestion
                                                {t('plugin_ito_claim_all_unlock_time', {
                                                    time: formatDateTime(swappedToken.unlockTime, 'yyyy-MM-dd hh:mm:ss')
                                                })}
```

Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx:163]:
```suggestion
                                                {t('plugin_ito_claim_all_unlock_time', {
                                                    time: formatDateTime(swappedToken.unlockTime, 'yyyy-MM-dd hh:mm:ss')
                                                })}
```

File: packages/maskbook/src/_locales/en/messages.json [hunk 4]
```diff
@@ -448,6 +448,10 @@
     "plugin_ito_dialog_claim_reminder_text4": "THE RIGHTS ARE NOT BEING OFFERED OR SOLD AND MAY NOT BE OFFERED OR SOLD, DIRECTLY OR INDIRECTLY, IN WHOLE OR IN PART, IN JURISDICTIONS OR TO WHOM SUCH AN OFFER WOULD BE UNLAWFUL.",
     "plugin_ito_expired": "Expired",
     "plugin_ito_create_new": "New",
+    "plugin_ito_claim_all_status_unclaimed": "Unclaimed",
+    "plugin_ito_claim_all_status_locked": "Locked",
+    "plugin_ito_claim_all_unlock_time": "Unlock Time",
```
Comment (septs) [inline packages/maskbook/src/_locales/en/messages.json:453]:
```suggestion
    "plugin_ito_claim_all_unlock_time": "Unlock Time: {{time}}",
```

File: packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx [hunk 5]
```diff
@@ -131,13 +147,21 @@ export function ClaimAllDialog(props: ClaimAllDialogProps) {
             }, 2000)
             return
         }
+        const claimableTokens = swappedTokens!.filter((t) => t.isClaimable)
+        const summary = claimableTokens.reduce((acc, cur, i) => {
+            const text = formatBalance(cur.amount, cur.token.decimals) + ' ' + cur.token.symbol
+            if (i === 0) return acc + ' ' + text
+            if (i === claimableTokens.length - 1) return acc + ' And ' + text
+            return acc + '、' + text
+        }, 'Claim')
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx:156]: see https://mdn.io/Intl.ListFormat

```
> new Intl.ListFormat('en').format(['A', 'B', 'C']);
< "A, B, and C"
> new Intl.ListFormat('zh-hant').format(['A', 'B', 'C']);
< "A、B和C"
```

Comment (zhouhanseng) [reply packages/maskbook/src/plugins/ITO/UI/ClaimAllDialog.tsx:156]: This is cool

**Review by septs — APPROVED** [review]

**Comment by AinuX** [comment]
Test passed

**Review by guanbinrui — APPROVED** [review]

**Review by septs — APPROVED** [review]

**Pull request merged**